packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -3785,4 +3785,334 @@ describe("typespec-client-generator-core: decorators", () => {
       strictEqual(method.operation.bodyParam.correspondingMethodParams[0], inputParam);
     });
   });
+  describe("@clientInitialization", () => {
+    it("main client", async () => {
+      await runner.compileWithCustomization(
+        `
+        @service
+        namespace MyService;
+
+        op download(@path blobName: string): void;
+        `,
+        `
+        namespace MyCustomizations;
+
+        model MyClientInitialization {
+          blobName: string;
+        }
+
+        @@clientInitialization(MyService, MyCustomizations.MyClientInitialization);
```

Comment:
So from TCGC's response perspective, there is not concept of `clientInitialization`, emitter just need to process the `SdkParameter` when process each operation. Does my understanding correct?

Reply:
```
    initialization: diagnostics.pipe(getSdkInitializationType(context, client)),
```

Reply:
@haolingdong-msft the model in `@clientInitialization` gets added as the `initialization` property on the `SdkClientType`. They are treated as method parameters, so we don't have a concept of them being query / body etc: that we can find out in `method.operation` from the params being mapped back to them